pkg/resources/resource_connection_aws.go

```diff
@@ -260,6 +261,130 @@ func connectionAwsUpdate(ctx context.Context, d *schema.ResourceData, meta inter
 		}
 	}
 
+	if d.HasChange("endpoint") {
+		oldEndpoint, newEndpoint := d.GetChange("endpoint")
+		b := materialize.NewConnection(metaDb, o)
+		if newEndpoint == nil || newEndpoint == "" {
+			if err := b.AlterDrop([]string{"ENDPOINT"}, validate); err != nil {
+				d.Set("endpoint", oldEndpoint)
+				return diag.FromErr(err)
+			}
+		} else {
+			options := map[string]interface{}{
+				"ENDPOINT": newEndpoint.(string),
+			}
+			if err := b.Alter(options, false, validate); err != nil {
+				d.Set("host", oldEndpoint)
```

Comment:
Should this be "endpoint"?

Reply:
Ah nice catch! This would have been a nasty bug if it had landed like that.